exonum-java-binding/core/rust/src/runtime/cmd.rs

```diff
@@ -15,150 +15,166 @@
-
-        let jvm_config = JvmConfig {
-            args_prepend,
-            args_append,
-            jvm_debug_socket,
-        };
-
-        let runtime_config = RuntimeConfig {
-            log_config_path,
-            port,
-            override_system_lib_path,
-        };
-
-        let config = Config {
-            jvm_config,
-            runtime_config,
-        };
-
-        // Write the final EJB configuration.
-        write_ejb_config(&mut context, Value::try_from(config)?);
-        Ok(context)
+/// Possible output of the Java Bindings CLI commands.
+pub enum EjbCommandResult {
+    /// Output of the standard Exonum Core commands.
+    Standard(StandardResult),
+    /// Output of EJB-specific `run` command.
+    EjbRun(Config),
+}
+
+impl From<StandardResult> for EjbCommandResult {
+    fn from(result: StandardResult) -> Self {
+        EjbCommandResult::Standard(result)
     }
 }
 
-/// Updates the `ejb` section of service configs of `NodeConfig` with `value` and puts updated
-/// `NodeConfig` back to `Context`.
-fn write_ejb_config(context: &mut Context, value: Value) {
-    let mut node_config = get_node_config(context);
-    node_config
-        .services_configs
-        .insert(EJB_CONFIG_SECTION_NAME.to_owned(), value);
-    context.set(keys::NODE_CONFIG, node_config);
+/// Interface of Java Bindings CLI commands.
+pub trait EjbCommand {
+    /// Returns the result of command execution.
+    fn execute(self) -> Result<EjbCommandResult, failure::Error>;
 }
 
-/// Extracts the `NodeConfig` from `Context` for further processing.
-fn get_node_config(context: &Context) -> NodeConfig<PathBuf> {
-    context
-        .get(keys::NODE_CONFIG)
-        .expect("Unable to read node configuration.")
+impl EjbCommand for Run {
+    fn execute(self) -> Result<EjbCommandResult, failure::Error> {
+        if let StandardResult::Run(node_run_config) = self.standard.execute()? {
+            let jvm_config = JvmConfig {
+                args_prepend: self.jvm_args_prepend,
+                args_append: self.jvm_args_append,
+                jvm_debug_socket: self.jvm_debug,
+            };
+
+            let log_config_path = self
+                .ejb_log_config_path
+                .unwrap_or_else(|| get_path_to_default_log_config());
+
+            let override_system_lib_path = self
+                .ejb_override_java_library_path
+                .map(|p| p.to_string_lossy().into_owned());
+
+            let runtime_config = RuntimeConfig {
+                artifacts_path: self.artifacts_path,
+                log_config_path,
+                port: self.ejb_port,
+                override_system_lib_path,
+            };
+
+            let config = Config {
+                run_config: node_run_config,
+                jvm_config,
+                runtime_config,
+            };
+
+            Ok(EjbCommandResult::EjbRun(config))
+        } else {
+            Err(format_err!("Standard run command returned invalid result"))
```

Comment:
Is there any context to include, if not already shown elsewhere?

Reply:
My fault, it is actually `unreachable`, I use `if let` for pattern-matching only. Fixed.